### PR TITLE
chore: fix anyio v4 testing related issues

### PR DESCRIPTION
**Pull request by Goldziher**
This PR fixes the testing issues caused by anyio 4, specifically our reliance on a private class that thas been removed.

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -106,8 +106,10 @@ uvicorn = { extras = ["standard"], version = ">=0.22.0", optional = true }
 aiosqlite = "*"
 asyncmy = "*"
 asyncpg = "*"
+beanie = ">=1.21.0"
```
Comment (euri10) [inline pyproject.toml:109]: not sure I understand how this is related with anyio
same for the changes [tests/unit/test_contrib/test_pydantic/test_beanie_integration.py](https://github.com/litestar-org/litestar/pull/2287/files#diff-fd74b705880373004fc0f91b302e044d7496cc47477c19be3c548e4244f7a62f)

**Review by euri10 — APPROVED** [review]
lgtm, not sure how some sqlalchemy tests fails on uuid with this though

File: pyproject.toml (hunk 1 above)
Comment (Goldziher) [reply pyproject.toml:109]: this was also failing in the pipeline, so I installed it. apparently it now supports pydantic v2, so no longer a need to drop it out.

File: tests/helpers.py [hunk 2]
```diff
@@ -38,7 +36,18 @@ async def maybe_async(obj: Awaitable[T] | T) -> T:
     return cast(T, await obj) if inspect.isawaitable(obj) else cast(T, obj)
 
 
+class _AsyncContextManagerWrapper(AsyncContextManager):
```
[the remark this answers is not in the document]
Comment (Goldziher) [reply tests/helpers.py:39]: sure

**Pull request merged**